dask/array/backends.py

```diff
@@ -296,3 +299,105 @@ def _nannumel_sparse(x, **kwargs):
     # If all dimensions are contracted, this will just be a number, otherwise we
     # want to densify it.
     return n.todense() if hasattr(n, "todense") else n
+
+
+class ArrayBackendEntrypoint(DaskBackendEntrypoint):
+    def __init__(self):
+        """Register data-directed dispatch functions"""
+        raise NotImplementedError
+
+    def default_random_state(self):
+        raise NotImplementedError
+
+    def new_random_state(self, state):
+        raise NotImplementedError
+
+    def ones(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def zeros(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def empty(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def full(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def arange(self, *args, **kwargs):
+        raise NotImplementedError
+
+
+array_creation_dispatch = CreationDispatch(
```

Comment:
In the design doc you say that this is intended to be *primarily* for IO-related functionality. What I see here is a mixture of things that imply IO and things that don't. That is to say `ArrayBackendEntrypoint` doesn't seem to be named as if it's only for IO (though its methods are currently IO-related), but the `array_creation_dispatch` function is.

What's your view on whether/how this will be extended to non-IO-related functionality? Are we planning for that case?

Reply:
I think the entrypoint should requre the definition of collection-creation functions, and should somehow ensure that the necessary compute-level dispatch functions are registered. I don't think that dask will ever support dispatching at the collection level for anything other than "creation." Once the collection is created, dispatching should be accomplished within the compute tasks, or through collection subclassing.